conn.go

```diff
@@ -166,6 +167,40 @@ type conn struct {
 	gss GSS
 }
 
+type syncErr struct {
+	err error
```

Comment:
any reason not to use `type syncErr atomic.Value` here, avoiding the mutex in the impls below?

Reply:
IIRC, I tried and ran into not being able to store a nil value. I currently don't see the code storing nil though, so I could have run into it while testing only. I'll give it a try.

Reply:
ah, the actual reason is that syncErr.set only wants to set an error if it is currently nil. atomic.Value's CompareAndSwap was introduced only in go1.17.